TECHNOLOGY_CHOICES.md

```diff
@@ -0,0 +1,104 @@
+# Technology Choices
+
+# Databases
+- PostgreSQL
+- Neo4J
+- CouchDB
+- eXist-db
+
+## Checks for determining database
+- Database should be able to store XML data
+- Database should be relational, because we need relations between files/pieces of files/models
+- Database must be provided with the solutions, for example part of a Docker image
+- Support must be available, in case of trouble.
+- Database must be open-source, because the whole solution must be open-source
+
+## PostgreSQL
+Homepage: https://www.postgresql.org/
+
+### Pros
+- Open Source (not entirely, working with patches)
+- Fully supports ACID
+
+### Cons
+- Does not support XML out-of-the-box
+
+## Neo4J
+
+## CouchDB
+
+## eXist-db
+Homepage: http://exist-db.org/exist/apps/homepage/index.html
+
+### Pros
+- Native XML database
+- Fully Open Source
+- Cross-platform
+- Active community (weekly community call, Slack channels, books)
+- Docker image available
+- Multiple query languages like HTTP, REST, xQuery and xPath
+- Can act like a graph database, so relations between xml fragments for example is possible.
+- Supports XML validation (https://exist-db.org/exist/apps/doc/validation.xml)
+
+### Cons
+- Using LGPL software is discouraged by Alliander when __modifying__ source code. So if we want to add features to eXist-db in the future, we might have a problem.
+- No SQL available (is this needed?)
+- Doesn't have all the ACID properties according to [vschart](http://vschart.com/compare/exist-db/vs/postgresql). Isolation is unknown, can't find other sources which confirm this.
+- Not well known, no clear use cases in production. There are some [here](http://showcases.exist-db.org/exist/apps/Showcases/index.html)
+- eXist needs JRE (Java Runtime Environment) to run
+
+# XML Processing
+## Checks for determining XML processing
+- Can manipulate/check XML configuration files by using rules
+- Can be embedded in our solution
+- Tool must be open source
+
+## RiseClipse
+### Pros
+- Main use is validating IEC 81850/IEC CIM configuration files, exactly what we need.
+- Usage experience within Alliander
+- [Docker image](https://hub.docker.com/r/riseclipse/riseclipse-validator-scl) available
+- Add own validation rules (in Object Constraint Language)
+
+### Cons
+- Development doesn't seem very active
+
+## Schematron
```

Comment:
Does it support OCL?

Reply:
It can do anything OCL can do:
https://shapechange.net/targets/xsd/extensions/ocl/

I've heard it can do more than OCL, but I cannot find those examples.